python/ray/serve/_private/controller.py

```diff
@@ -375,6 +384,109 @@ def _update_proxy_nodes(self):
+                    min_repl_adj = snapshot["min_replicas"]
+                    max_repl_adj = snapshot["max_replicas"]
+                    total_requests = snapshot["total_requests"]
+
+                    if isinstance(autoscaling_config, dict):
+                        policy_name = autoscaling_config.get("policy", "default")
+                        look_back_period_s = autoscaling_config.get(
+                            "look_back_period_s"
+                        )
+                    else:
+                        policy_name = getattr(autoscaling_config, "name", "default")
+                        look_back_period_s = getattr(
+                            autoscaling_config, "look_back_period_s", None
+                        )
+
+                    if proposed_replicas > current_replicas:
+                        scaling_status = "UPSCALING"
+                    elif proposed_replicas < current_replicas:
+                        scaling_status = "DOWNSCALING"
+                    else:
+                        scaling_status = "STABLE"
+                    decision = ScalingDecision(
+                        timestamp_s=time.time(),
+                        reason=f"current={current_replicas}, proposed={proposed_replicas}",
+                        prev_num_replicas=int(current_replicas),
+                        curr_num_replicas=int(proposed_replicas),
+                        policy=policy_name,
+                    )
+
+                    self._scaling_decisions.setdefault(dep_id, []).append(decision)
+                    if len(self._scaling_decisions[dep_id]) > 50:
+                        self._scaling_decisions[dep_id] = self._scaling_decisions[
+                            dep_id
+                        ][-50:]
+                    key = (app_name, dep_name)
+                    signature = self._serve_event_summarizer.compute_signature(
+                        current_replicas=current_replicas,
+                        proposed_replicas=proposed_replicas,
+                        min_replicas=min_repl_adj,
+                        max_replicas=max_repl_adj,
+                        scaling_status=scaling_status,
+                        total_requests=total_requests,
+                    )
+                    if self._last_autoscaling_snapshots.get(key) == signature:
+                        continue
+                    recent_decisions = self._scaling_decisions.get(dep_id, [])
+                    decisions_summary = (
+                        self._serve_event_summarizer.summarize_recent_decisions(
+                            recent_decisions, limit=2
+                        )
+                    )
+                    scaling_status_h = (
+                        self._serve_event_summarizer.format_scaling_status(
+                            scaling_status
+                        )
+                    )
+                    payload = self._serve_event_summarizer.build_snapshot_payload(
```

Comment:
do you expect payload to be used for anything other than `log_snapshot`? Why not log it as part of same function?